programs/client/BuzzHouse/Generator/ExternalIntegrations.h

```diff
@@ -0,0 +1,953 @@
+#pragma once
+
+#include "config.h"
+
+#include "FuzzConfig.h"
+#include "SQLCatalog.h"
+
+#if defined USE_MYSQL && USE_MYSQL
+#    if __has_include(<mysql.h>)
+#        include <mysql.h>
+#    else
+#        include <mysql/mysql.h>
+#    endif
+#endif
+
+#if defined USE_MONGODB && USE_MONGODB
+#    include <bsoncxx/builder/stream/array.hpp>
+#    include <bsoncxx/builder/stream/document.hpp>
+#    include <bsoncxx/json.hpp>
+#    include <bsoncxx/types.hpp>
+#    include <mongocxx/client.hpp>
+#    include <mongocxx/collection.hpp>
+#    include <mongocxx/database.hpp>
+#endif
+
+#if defined USE_LIBPQXX && USE_LIBPQXX
+#    include <pqxx/pqxx>
+#endif
+
+#if defined USE_SQLITE && USE_SQLITE
+#    include <sqlite3.h>
+#endif
```

Comment:
I think there's no need to check whether things are defined or not, because they should always be defined in the config.h

```suggestion
#if USE_MYSQL
#    if __has_include(<mysql.h>)
#        include <mysql.h>
#    else
#        include <mysql/mysql.h>
#    endif
#endif

#if USE_MONGODB
#    include <bsoncxx/builder/stream/array.hpp>
#    include <bsoncxx/builder/stream/document.hpp>
#    include <bsoncxx/json.hpp>
#    include <bsoncxx/types.hpp>
#    include <mongocxx/client.hpp>
#    include <mongocxx/collection.hpp>
#    include <mongocxx/database.hpp>
#endif

#if USE_LIBPQXX
#    include <pqxx/pqxx>
#endif

#if USE_SQLITE
#    include <sqlite3.h>
#endif
```

Reply:
Sometimes, they may be defined as 0, so double-checking is good.

Reply:
>Sometimes, they may be defined as 0, so double-checking is good.

Of course you need to check whether they're 0 or not, but you don't need to check if they're defined. They're always defined, and their value is either 1 or 0.

Reply:
From here: https://stackoverflow.com/questions/17160755/is-there-a-way-to-both-check-a-macro-is-defined-and-it-equals-a-certain-value-at there's no straight way to do it